src/main/java/bio/terra/service/iam/sam/SamIam.java

```diff
@@ -133,6 +142,7 @@ private boolean isAuthorizedInner(AuthenticatedUserRequest userReq,
         try (Stream<ResourceAndAccessPolicy> resultStream =
                  samResourceApi.listResourcesAndPolicies(iamResourceType.toString()).stream()) {
             return resultStream
+                .filter(resource -> ValidationUtils.isValidUUID(resource.getResourceId()))
                 .map(resource -> UUID.fromString(resource.getResourceId()))
```

Comment:
Slight improvement (?)
```suggestion
            return resultStream
                .map(ResourceAndAccessPolicy::getResourceId)
                .filter(ValidationUtils::isValidUUID)
                .map(UUID::fromString)
```

This is a tiny bit wasteful in that it calls `UUID.fromString` twice. If you had a util function that returned `Optional<UUID>` you could `flatMap()` it, I think. Might be better than isValid/fromString.

Reply:
I think that's a great idea.  will do